ui/src/components/ErrorPanel/index.jsx

```diff
@@ -97,25 +97,57 @@ export default class ErrorPanel extends Component {
     } = this.props;
     const { error } = this.state;
     const hasWarning = Boolean(props.warning);
-    const errorMessage =
-      error && error.graphQLErrors && error.graphQLErrors[0]
-        ? error.graphQLErrors[0].message
-        : error;
+    const message = [];
+
+    if (!error) {
+      return null;
+    }
+
+    // handle GraphQLErrors as a special case
+    if (error.graphQLErrors && error.graphQLErrors.length > 0) {
+      const errors = error.graphQLErrors;
+
+      // Log extensions for all errors received
+      errors.forEach(err => {
+        if (err?.extensions) {
+          // eslint-disable-next-line no-console
+          console.log('Error from web-server:', err.extensions);
+        }
+      });
+
+      // construct a markdown summary of all of the errors
+      if (errors.length > 1) {
+        message.push('Multiple errors from web-server:');
+
+        errors.forEach(err => {
+          message.push(`* ${err.message}`);
+        });
```

Comment:
In the past we would only show the first error to avoid having a very massive error panel show up when the user is missing a lot of scopes. Would it still make sense to only show the first error or perhaps put a limit on how many messages we can render from this array (e.g., only show 2/3 messages at most)?

Reply:
Good point.  I'll limit it to 3 with an "and N additional errors".